hca/util/__init__.py

```diff
@@ -198,7 +199,29 @@ def iterate(self, **kwargs):
     def paginate(self, **kwargs):
         """Yield paginated responses one response body at a time."""
         for page in self._get_raw_pages(**kwargs):
-            yield page.json()
+            data = page.json()
+            content_key = page.headers.get("X-OpenAPI-Paginated-Content-Key", "results")
+            data['_internal_api_content_key'] = content_key
```

Comment:
I would prefer it if we could find a way not to pass this in-band.

Reply:
I think an alternative way to perform this would be to populate the header value in the `swagger.json` page, then create a internal function that can extract the value from 
`_PaginatingClientMethodFactory.method_data`. This would require changes to the DSS Swagger, and potentially Fusillade/Query Service Swagger pages as well.

Reply:
You can use `_PaginatingClientMethodFactory. _get_raw_pages()`, the method `paginate()` wraps. `_get_raw_pages()` yields entire responses. You can read the paginated content key from the response headers.